connectors/src/connectors/microsoft/temporal/workflows.ts

```diff
@@ -47,9 +50,21 @@ export async function fullSyncWorkflow({
 }) {
   await syncStarted(connectorId);
 
-  if (nodeIdsToSync === undefined) {
-    nodeIdsToSync = await getRootNodesToSync(connectorId);
-  }
+  setHandler(folderUpdatesSignal, (folderUpdates: FolderUpdatesSignal[]) => {
+    // If we get a signal, update the workflow state by adding/removing folder ids.
+    for (const { action, folderId } of folderUpdates) {
+      switch (action) {
+        case "added":
```

Comment:
I have multiple questions here:
- populateDeltas not executed for added node?
- if node is added during the syncSucceed activity, node will be skipped?
- if node is removed while it was already traversed by the while loop, then it will splice(-1,1) which will remove the last element? (not what we'd want)

Reply:
maybe on restart we could somehow restart the workflow with the right nodeIdsToSync, so it would avoid issues above + the need to skip of previous comments?

Reply:
also if folders were already synced how are they removed? I think you had told me but can't remember

Reply:
- populateDeltas is called directly in setPermissions. I don't know what's the use of the 2nd populateDelta in the workflow - that one will indeed not be called when adding nodes.
- if node is added during the syncSucceed : I've added a check after the syncSucceeded, in case it happens
- fixed the splice. on the other hand, I notice that "removed" signal is never sent ... I implemented it in a first version but finally only rely on the removal done in setPermissions(). btw i think we have the same issue with google connector. We should probably get rid of the microsoftDeletionWorkflow and handle all deletion inside the fullSync.

Reply:
Ok but in that case with panic flag set to `true` right?
Because throwing, by making the activity retry, was triggering a monitor if the issue wasn't transient, that was prompting us to act; this is why I was inclined to keep the throw